components/layout/layout_task.rs

```diff
@@ -865,6 +874,114 @@ impl LayoutTask {
+        // There are probably other quirks.
+        let applies = true;
+
+        // TODO: we will return neither the computed nor used value for margin and padding.
+        // Firefox returns blank strings for the computed value of shorthands,
+        // so this should be web-compatible.
+        match property.clone() {
+            atom!("margin-bottom") | atom!("margin-top") |
+            atom!("margin-left") | atom!("margin-right") |
+            atom!("padding-bottom") | atom!("padding-top") |
+            atom!("padding-left") | atom!("padding-right")
+            if applies && style.get_box().display != display::computed_value::T::none => {
+                let (margin_padding, side) = match *property {
+                    atom!("margin-bottom") => (MarginPadding::Margin, Side::Bottom),
+                    atom!("margin-top") => (MarginPadding::Margin, Side::Top),
+                    atom!("margin-left") => (MarginPadding::Margin, Side::Left),
+                    atom!("margin-right") => (MarginPadding::Margin, Side::Right),
+                    atom!("padding-bottom") => (MarginPadding::Padding, Side::Bottom),
+                    atom!("padding-top") => (MarginPadding::Padding, Side::Top),
+                    atom!("padding-left") => (MarginPadding::Padding, Side::Left),
+                    atom!("padding-right") => (MarginPadding::Padding, Side::Right),
+                    _ => unreachable!()
+                };
+                let requested_node: OpaqueNode = OpaqueNodeMethods::from_script_node(requested_node);
+                let mut iterator =
+                    MarginRetrievingFragmentBorderBoxIterator::new(requested_node,
+                                                                   side,
+                                                                   margin_padding,
+                                                                   style.writing_mode);
+                sequential::iterate_through_flow_tree_fragment_border_boxes(layout_root, &mut iterator);
+                rw_data.resolved_style_response = iterator.result.map(|r| r.to_css_string());
+            },
+
+            atom!("bottom") | atom!("top") | atom!("right") |
+            atom!("left") | atom!("width") | atom!("height")
+            if applies && positioned && style.get_box().display != display::computed_value::T::none => {
+                let layout_data = layout_node.borrow_layout_data();
+                let position = layout_data.as_ref().map(|layout_data| {
+                    match layout_data.data.flow_construction_result {
+                        ConstructionResult::Flow(ref flow_ref, _) => {
+                            let base = flow::base(flow_ref.deref());
+                            base.stacking_relative_position
+                        }
+                        _ => panic!("tried to resolve style of element without flow ref"),
```

Comment:
This is gonna cause panics whenever you call gCS on inlines. Can you put a TODO in here and return some bogus value?